### PR TITLE
Added an empty placeholder to select a city

### PR DESCRIPTION
**Pull request by valko-pro**

### CODE REVIEW

**Comment by valko-pro** [comment]
image

File: front/assets/js/main.js [hunk 1]
```diff
@@ -32,6 +33,7 @@
 					$( '#shipping_nova_poshta_for_woocommerce_warehouse' ).addClass( 'inactive' );
 				},
 				success: function( data ) {
+					warehouse_wrapper.show();
```
Comment (wppunk) [inline front/assets/js/main.js:36]: move it to the next line after variables.

File: admin/class-user.php [hunk 2]
```diff
@@ -71,18 +71,6 @@ public function fields() {
...
-					$user_id,
-					$this->language->get_current_language()
-				),
-				1
-			);
-			$city_id = array_keys( $city )[0];
-			$city    = array_pop( $city );
 		}
 
 		$warehouses   = $this->api->warehouses( $city_id );
```
Comment (wppunk) [inline admin/class-user.php:76]: What If empty `$city_id`?

File: admin/class-user.php [hunk 3]
```diff
@@ -96,20 +84,21 @@ public function fields() {
...
+				'placeholder' => __( 'Choose city', 'shipping-nova-poshta-for-woocommerce' ),
 			],
 
 			'shipping_nova_poshta_for_woocommerce_warehouse' => [
 				'type'     => 'select',
 				'label'    => __( 'Choose branch', 'shipping-nova-poshta-for-woocommerce' ),
 				'required' => true,
-				'options'  => $warehouses,
-				'default'  => $warehouse_id,
+				'options'  => [ '' => '' ],
```
Comment (wppunk) [inline admin/class-user.php:100]: What for customers who have user_meta with the current warehouse? I don't want to choose these fields again :)

File: front/assets/css/main.css [hunk 4]
```diff
@@ -13,3 +13,17 @@
     cursor: no-drop;
     opacity: .75;
 }
+
+#shipping_nova_poshta_for_woocommerce_warehouse_field {
```
Comment (wppunk) [inline front/assets/css/main.css:17]: Try to use:
```
#shipping_nova_poshta_for_woocommerce_city_field .select2-container,
#shipping_nova_poshta_for_woocommerce_warehouse_field .select2-container {
	display: block;
	width: 100%;
}
```

**Review by wppunk — CHANGES_REQUESTED** [review]

File: front/assets/js/main.js (hunk 1 above)
Comment (valko-pro) [reply front/assets/js/main.js:36]: Okay

File: admin/class-user.php (hunk 2 above)
Comment (valko-pro) [reply admin/class-user.php:76]: You said that we have city id always )))

File: admin/class-user.php (hunk 3 above)
Comment (valko-pro) [reply admin/class-user.php:100]: Okay

File: front/assets/css/main.css (hunk 4 above)
Comment (valko-pro) [reply front/assets/css/main.css:17]: And max-width)

File: admin/class-user.php [hunk 5]
```diff
@@ -63,25 +63,28 @@ public function hooks() {
 	 * TODO: Move to other place.
 	 */
 	public function fields() {
-		$user_id      = get_current_user_id();
+		$user_id      = is_user_logged_in() ? get_current_user_id() : '';
 		$city_id      = filter_input( INPUT_POST, 'shipping_nova_poshta_for_woocommerce_city', FILTER_SANITIZE_STRING );
 		$warehouse_id = filter_input( INPUT_POST, 'shipping_nova_poshta_for_woocommerce_warehouse', FILTER_SANITIZE_STRING );
```
Comment (wppunk) [inline admin/class-user.php:68]: Add `$warehouse_classes = [];`

File: admin/class-user.php [hunk 6]
```diff
@@ -63,25 +63,28 @@ public function hooks() {
...
 		if ( empty( $city_id || $warehouse_id ) ) {
 			$city_id = apply_filters( 'shipping_nova_poshta_for_woocommerce_default_city_id', '', $user_id );
 		}
```
Comment (wppunk) [inline admin/class-user.php:71]:
```
$warehouse_id      = apply_filters(
	'shipping_nova_poshta_for_woocommerce_default_warehouse_id',
	$warehouse_id,
	$user_id,
	$city_id
);
```

Comment (wppunk) [reply admin/class-user.php:71]: `$warehouses   = [ '' => '' ];`

File: admin/class-user.php [hunk 7]
```diff
@@ -63,25 +63,28 @@ public function hooks() {
...
 		if ( $city_id ) {
-			$city = $this->api->city( $city_id );
+			$city              = $this->api->city( $city_id );
+			$warehouses        = $this->api->warehouses( $city_id );
+			$warehouse_id      = array_keys( $warehouses )[0] ?? '';
```
Comment (wppunk) [inline admin/class-user.php:75]: Remove `$warehouse_id` from this `if`

File: admin/class-user.php [hunk 8]
```diff
@@ -63,25 +63,28 @@ public function hooks() {
...
+			$warehouse_id      = apply_filters(
+				'shipping_nova_poshta_for_woocommerce_default_warehouse_id',
+				$warehouse_id,
+				$user_id,
+				$city_id
+			);
+			$warehouse_classes = 'has-meta';
+		} else {
```
Comment (wppunk) [inline admin/class-user.php:83]: Remove `else`

File: admin/class-user.php [hunk 9]
```diff
@@ -97,9 +100,10 @@ public function fields() {
 				'type'     => 'select',
 				'label'    => __( 'Choose branch', 'shipping-nova-poshta-for-woocommerce' ),
 				'required' => true,
-				'options'  => [ '' => '' ],
-				'default'  => '',
+				'options'  => $warehouses,
+				'default'  => $warehouse_id,
```
Comment (wppunk) [inline admin/class-user.php:104]: `isset( $warehouses [ $warehouse_id ] ) ? $warehouse_id : '';`

**Review by wppunk — CHANGES_REQUESTED** [review]
Check my comments

**Pull request closed without merging**